tests/product/test_server_install.py

```diff
@@ -171,6 +171,50 @@ def assert_installed_with_regex_configs(self, master, slaves):
                                            '/etc/presto/config.properties',
                                            self.default_workers_config_regex_)
 
+    def test_install_with_java8_home(self, dummy=True):
+        for container in self.cluster.all_hosts():
+            self.cluster.exec_cmd_on_host(container,
+                                          "mv /usr/java/jdk1.8.0_40 /usr/")
+        topology = {"coordinator": "master",
+                    "workers": ["slave1", "slave2", "slave3"],
+                    "java8_home": "/usr/jdk1.8.0_40/jre"}
+        self.upload_topology(topology)
+
+        cmd_output = self.installer.install(dummy)
+        expected = installed_all_hosts_output
+
+        actual = cmd_output.splitlines()
+        self.assertEqual(sorted(expected), sorted(actual))
+
+        for container in self.cluster.all_hosts():
+            self.installer.assert_installed(self, container)
+            self.assert_has_default_config(container)
+            self.assert_has_default_connector(container)
+
+    def test_install_failure_without_java8_home(self, dummy=True):
+        for container in self.cluster.all_hosts():
+            self.cluster.exec_cmd_on_host(container,
+                                          "mv /usr/java/jdk1.8.0_40 /usr/")
+        self.upload_topology()
+        cmd_output = self.installer.install(dummy)
+        actual = cmd_output.splitlines()
+        num_failures = 0
+        for line in enumerate(actual):
+            if str(line).find("Error: Required Java version"
+                              " could not be found") != -1:
+                num_failures += 1
+
+        self.assertEqual(4, num_failures)
+
+        no_rpm_installs = 0
+        for container in self.cluster.all_hosts():
+            output = self.cluster.exec_cmd_on_host(container, "rpm -qa")
+            outputrpms = output.splitlines()
+            if "dummy-rpm.rpm" not in outputrpms:
+                no_rpm_installs += 1
+
+        self.assertEquals(4, no_rpm_installs)
```

Comment:
Changed, used assert_uninstalled(), also modified assert_uninstalled() to support dummy rpm uninstall.